inventory/src/main/java/com/yas/inventory/service/StockService.java

```diff
@@ -50,4 +54,38 @@ public void addProductIntoWarehouse(List<StockPostVM> postVMs) {
 
     }
 
+    public void checkQuantityAvailable(List<StockCheckRequest> stockCheckRequests) {
+        List<Long> productIds = stockCheckRequests.stream().map(StockCheckRequest::productId).toList();
+        List<Stock> stocks = stockRepository.getAvailableStockByProductIds(productIds);
+        List<OutStockErrorVm> errors = new ArrayList<>();
+        stockCheckRequests.forEach(scr -> {
+            List<Stock> stockChecks = stocks.stream()
+                    .filter(stock -> scr.productId().equals(stock.getProductId())).toList();
+            if (!isValidOrderQuantity(stockChecks, scr.quantity())) {
+                errors.add(new OutStockErrorVm(scr.productId(), scr.quantity()));
+            }
+        });
+
+        if (!errors.isEmpty()) {
+            throw new OutOfStockException(MessageCode.OUT_OF_STOCK, errors);
+        }
+        stockRepository.saveAll(stocks);
+    }
```

Comment:
I also think it should return 1 stock or warehouse so we can use it later to reduce the quantity when order is success so that so we don't need to check which warehouse is available again reduce the quantity, should we?

Reply:
No, we don't need to return WH or stock, because we have use reserved quantity, and when reduce quantity, we just find reserved quantity > 0 and quantity > 0 in stock and execute to reduce quantity.

Reply:
I will update, thanks.